### PR TITLE
Extract relevant functions from node-nlp

### PR DESCRIPTION
**Pull request by Scar26**
I stripped down the node-nlp package and extracted only the parts that we're using.
12.5% decrease in storage space after running npm install

before:
before
after:
after

### CODE REVIEW

**Comment by bkimminich** [comment]
Hm, while I of course appreciate saving 20 MB of unneeded code dependencies, I would expect that this doesn't fully transfer to reduced Docker image size, or does it? But we are kind of setting us up for additional maintenance effort b/c we'd now have to keep all those `@nlp`-dependencies in harmony ourselves.

On the other hand, we might find so vulnerability in one of the older versions of said sub-dependencies and could pin it for an extra challenge in the future, but I'm not sure this is  worth it either.

@J12934, please help me out with a second opinion... 🤔

**Comment by Scar26** [comment]
Yes, I also agree that this 20mb decrease probably won't lead to an equivalent change in the sizer of the docker image, as it only went up by around 60mb after the integration of this module.

As for `@nlp` packages, I don't think we would need to do upgrade them regularly since they work pretty much independently anyway. We could just keep them as is until a new node version comes out or someone points out an unintended vulnerability in one of them. But yes, in both these cases the upgrade would be a little more painful than other packages if they introduce any structural changes.

The only sustainable solution I can think of right now, that would also minimize bloat is to do away with these libraries all together and write a small sentiment analyzer manually

**Comment by J12934** [comment]
Will try it out later.
Image size is (at least for me) only the secondary factor. Once the image is downloaded it can be cached on the node.

More important is, at least for me, how much RAM every Juice Shop instance requires to run. If this could decrease this, even by a few MB this would be great 👍

Will try to test this out this / tomorrow evening.

**Comment by bkimminich** [comment]
If RAM usage now went down that would mean NLP loads all kinds of unused stuff into memory... That'd be weird, but let's see what your trial shows! 👍

**Comment by J12934** [comment]
Looks pretty promising 👏🚀:

## Docker Image Size:

Screenshot 2020-11-16 at 21 09 23

## RAM Usage (No Load)

<table>
<tr>
	<td> Screenshot 2020-11-16 at 21 18 09
	<td> Screenshot 2020-11-16 at 21 23 05
</table>

File: nlp/index.js [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+      if (this.settings.ner.ducklingUrl || this.settings.ner.useDuckling) {
+        const builtin = new BuiltinDuckling(this.settings.ner)
+        this.container.register('extract-builtin-??', builtin, true)
+      } else {
+        const builtin = new BuiltinMicrosoft(this.settings.ner)
+        this.container.register('extract-builtin-??', builtin, true)
+      }
+    } else {
+      const builtin = new BuiltinMicrosoft(this.settings.ner)
+      this.container.register('extract-builtin-??', builtin, true)
```
Comment (J12934) [inline nlp/index.js:63]: This `BuiltinDuckling` and `BuiltinMicrosoft` seem to be mutually exclusive, are we using both?
Or could we potentially get rid of one of them?

Comment (J12934) [reply nlp/index.js:63]: Took a look at it, seems like we're using the default microsoft one which is way way larger than the other one, so getting rid of the other would probably not really improve much.

Comment (Scar26) [reply nlp/index.js:63]: I just tested this, and we can actually get rid of both!

Comment (Scar26) [reply nlp/index.js]: pushed. 
Storage also went down by another 20 mb
after2

Comment (J12934) [reply nlp/index.js]: Docker Image size is also quite a bit down (to `415MB`)
So all in all this brings it down from `455MB` to `415MB`

**Review by J12934 — APPROVED** [review]

File: nlp/index.js (hunk 1 above)
Comment (J12934) [reply nlp/index.js]: Wait no, memory usage also improved a bit:

Screenshot 2020-11-16 at 22 34 04

**Comment by bkimminich** [comment]
Awesome, then let's do this! 🥳👍

**Pull request merged**